src/odemis/dataio/hdf5.py

```diff
@@ -1663,3 +1647,14 @@ def read_thumbnail(filename):
     # TODO: support filename to be a File or Stream
 
     return _thumbFromHDF5(filename)
+
+
+def read_str(s):
```

Comment:
As the function doesn't really read, but just converts, it's a little bit misnamed. I'd suggest `ensure_str` or `convert_to_str`
Also, please add typing info:
`(s: Union[bytes, str]) -> str:`